scipy/spatial/_plotutils.py

```diff
@@ -94,20 +94,25 @@ def convex_hull_plot_2d(hull, ax=None):
     Requires Matplotlib.
 
     """
+    from matplotlib.collections import LineCollection
+
     if hull.points.shape[1] != 2:
         raise ValueError("Convex hull is not 2-D")
 
     ax.plot(hull.points[:,0], hull.points[:,1], 'o')
+    line_segments = []
     for simplex in hull.simplices:
-        ax.plot(hull.points[simplex,0], hull.points[simplex,1], 'k-')
-
+        line_segments.append([(x, y) for x, y in hull.points[simplex]])
+    ax.add_collection(LineCollection(line_segments,
+                                     colors='k',
+                                     linestyle='solid'))
     _adjust_bounds(ax, hull.points)
 
     return ax.figure
 
 
 @_held_figure
-def voronoi_plot_2d(vor, ax=None):
+def voronoi_plot_2d(vor, ax=None, **kw):
```

Comment:
I tried it as you suggest initially, but the ``_held_figure`` decorator was raising the following error:
```
TypeError: _held_figure() takes from 2 to 3 positional arguments but 4 were given
```